src/ArmTemplates/Extractor/Utilities/AzureCliAuthenticator.cs

```diff
@@ -19,7 +19,7 @@ public class AzureCliAuthenticator
         string internalAzToken;
         string internalAzSubscriptionId;
 
-        public async Task<(string azToken, string azSubscriptionId)> GetAccessToken()
+        public virtual async Task<(string azToken, string azSubscriptionId)> GetAccessToken()
```

Comment:
For this one I would make an interface actually. Or just register it in DI and pass it in constructor as a class.
Again making a method virtual, only for possibility of Moq to access it in reflection is not a best idea IMO.

But it still works, so if you think in another way, we can ask Tom or Milan or anybody else about this

Reply:
I believe both of the approaches are for adding possibility to rewrite/reimplement the AzureClieAuthenticator
1 - add virtual to allow mock
2 - create interface to allow mock

I went with 1 approach for couple of reasong:
1 - AzureClieAuthenticator in my opinion will not have additional implementation or different approaches in near future. 
2 - Less code base changes

Reply:
Thanks for introducing this in another PR, and sorry for delaying the development a little bit